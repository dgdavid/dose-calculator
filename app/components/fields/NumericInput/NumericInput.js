import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, TextInput, View } from 'react-native';

import colors from '../../../config/colors';

const NumericInput = ({ unit, unitStyle, style, registerInput, ...props }) => (
  <View style={styles.wrapper}>
    <TextInput
      selectTextOnFocus
      keyboardType="numeric"
      underlineColorAndroid="transparent"
      ref={(input) => registerInput && registerInput(input)}
      style={[styles.input, style, props.editable === false && styles.disabled]} // eslint-disable-line
      {...props}
    />
    <Text style={unitStyle}>{unit}</Text>
  </View>
);

NumericInput.propTypes = {
  style: TextInput.propTypes.style,
  unit: PropTypes.string.isRequired,
  unitStyle: Text.propTypes.style,
  registerInput: PropTypes.func,
};

NumericInput.defaultProps = {
  unitStyle: null,
};

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },

  input: {
    color: colors.materialBlueGray500,
    fontSize: 40,
    paddingBottom: 0,
    textAlign: 'center',
  },

  disabled: {
    color: colors.materialBlueGray300,
  },
});

export default NumericInput;
