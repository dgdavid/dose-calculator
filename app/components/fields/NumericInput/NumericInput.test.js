import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NumericInput from './NumericInput';

test('render correctly', () => {
  const tree = renderer.create(<NumericInput unit={'ml'} />).toJSON();

  expect(tree).toMatchSnapshot();
});
