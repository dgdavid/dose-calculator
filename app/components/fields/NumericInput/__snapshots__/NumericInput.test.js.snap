// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    selectTextOnFocus={true}
    style={
      Array [
        Object {
          "color": "#607D8B",
          "fontSize": 40,
          "paddingBottom": 0,
          "textAlign": "center",
        },
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    ml
  </Text>
</View>
`;
